### PR TITLE
Support multiple versions for the documentation

### PR DESCRIPTION
**Pull request by lfades**
Review of changes:

- The documentation will come from the latest released tag, instead of `canary`, and it will use `master` if for some reason the fetch for the tag failed.
- The documentation page will now accept a custom tag, and will fetch the docs from that tag instead, valid examples for paths are:
  - `/docs/getting-started`
  - `/docs/tag/v9.2.2/getting-started` 
  - `/docs/tag/v9.2.1-canary.10/getting-started`
  - `/docs/tag/canary/getting-started` 
  - `/docs/tag/master/getting-started` 
- With the above, it would be possible to introduce a version switcher in the future, relative links, and navigations will stay in the same version. We could also preview new documentation changes using `/docs/tag/canary`.
- The edit button at the bottom of every documentation page will switch to "Go to the live version of this page" if a tag is being used.
- The edit button will always go to `canary`.
- In the case of a version not having a routes manifest (released before the new docs), it will show a 404.

Bug fixes:
- Fixed issue with relative URLs to non documentation files inside the readmes in the Next.js repo.
- Documentation paths will no longer 404 if there are uppercase characters

### CODE REVIEW

**Comment by zeit-team** [comment]
#### 📝Changed routes:


| [**/docs/\[...slug\]**](https://next-site-git-fetch-docs-stable.zeit.now.sh/docs/[...slug]) |
|:-:|
|<a href="https://next-site-git-fetch-docs-stable.zeit.now.sh/docs/[...slug]">Screenshot of /docs/[...slug]</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/OWKWjdGkb7STCuxMvzs_aJHLWDw_.png">(view full size)</a> |







Commit 968ff89c69eb05979e7a50037b0b4735c34a2681 (https://next-site-git-fetch-docs-stable.zeit.now.sh).

File: pages/docs/[...slug].js [hunk 1]
```diff
@@ -122,20 +124,26 @@ const Docs = ({ routes, route, data, html }) => {
...
-  const manifest = await fetchDocsManifest();
-  const route = findRouteByPath(slug, manifest.routes);
+  const { tag, slug } = getSlug(params);
+  const currentTag = tag || (await getLatestTag());
+  const manifest = await fetchDocsManifest(currentTag).catch(error => {
+    // If a manifest wasn't found for a custom tag, show a 404 instead
+    if (error.status === 404) return;
+    throw error;
+  });
+  const route = manifest && findRouteByPath(slug, manifest.routes);
```
Comment (ijjk) [inline pages/docs/[...slug].js]: We need to remove the `tag/vX.X.X` from the slug before looking up the route right?

File: components/docs/docs-page.js [hunk 2]
```diff
@@ -54,20 +58,21 @@ function DocsPage({ route, routes, html }) {
...
         .docs {
-          max-width: 100%;
-        }
-        @media screen and (max-width: 950px) {
-          .docs {
-            max-width: 100%;
-            margin: 0;
-          }
+          /* 300px is the sidebar width and its margin */
+          min-width: calc(100% - 300px - 1rem);
```
Comment (chibicode) [inline components/docs/docs-page.js]: Why is this being changed?

File: pages/docs/[...slug].js [hunk 3]
```diff
@@ -55,20 +56,24 @@ function SidebarRoutes({ isMobile, routes: currentRoutes, level = 1 }) {
...
     const route = { href, path, title, pathname, selected };
 
     return <Post key={title} isMobile={isMobile} level={level} route={route} />;
   });
 }
 
 const Docs = ({ routes, route, data, html }) => {
+  const router = useRouter();
+
+  console.log('DATA', { routes, route, router });
```
Comment (chibicode) [inline pages/docs/[...slug].js]: Let's remove this now

File: pages/docs/[...slug].js [hunk 4]
```diff
@@ -128,21 +133,36 @@ const Docs = ({ routes, route, data, html }) => {
...
+  const route = manifest && findRouteByPath(slug, manifest.routes);
+
+  console.log('SSG', {
+    params,
+    tag,
+    slug,
+    currentTag,
+    route,
+    hasRoutes: Boolean(manifest && manifest.routes)
+  });
```
Comment (chibicode) [inline pages/docs/[...slug].js]: Let's remove this now also

**Comment by chibicode** [comment]
Also checked that showcase pages work correctly. https://next-site-git-fetch-docs-stable.zeit.now.sh/showcase/

File: pages/docs/[...slug].js [hunk 5]
```diff
@@ -55,20 +56,22 @@ function SidebarRoutes({ isMobile, routes: currentRoutes, level = 1 }) {
...
 
     return <Post key={title} isMobile={isMobile} level={level} route={route} />;
   });
 }
 
 const Docs = ({ routes, route, data, html }) => {
+  const router = useRouter();
+
   if (!route) {
     return <Error statusCode={404} />;
```
Comment (chibicode) [inline pages/docs/[...slug].js]: Right now it's showing 404 as a skeleton for ungenerated pages - we need to make the page show something even if `routes`, `route`, `data`, and `html` are missing.

**Comment by chibicode** [comment]
Can we also create `public/robots.txt` and make sure tags are not crawled?

```
User-agent: *
Disallow: 
Disallow: /docs/tag/
```

We want to avoid what happened to Bootstrap - Google crawled v4.0 (older version) docs of Bootstrap docs and that older version comes higher on search results than the latest version. We can avoid this by not having Google index tagged versions.

Screen Shot 2020-02-28 at 5 08 47 PM

**Pull request merged**